File: workers/cartographer/testdata/sample.cpp
// Sample C++ file for Cartographer AST extraction tests.

#include <string>

struct Config {
    std::string path;
    int max_size;
};

typedef struct {
    std::string name;
} Record;

class DataProcessor {
public:
    explicit DataProcessor(const Config& config) : config_(config) {}

    std::string process(const std::string& record) {
        return record;
    }

private:
    Config config_;
};

Config load_config(const std::string& path) {
    Config cfg;
    cfg.path = path;
    cfg.max_size = 1024;
    return cfg;
}

File: workers/cartographer/testdata/sample.py
"""Sample Python file for Cartographer AST extraction tests."""


class DataProcessor:
    """Processes incoming data records."""

    def process(self, record):
        """Process a single record and return it."""
        return record

    def validate(self, record):
        """Validate a record before processing."""
        return bool(record)


class Config:
    """Configuration container."""

    def __init__(self, path):
        self.path = path


def load_config(path):
    """Load configuration from disk."""
    return Config(path)


def _internal_helper():
    """Private helper — should still be extracted (Python has no enforced access)."""
    pass
